src/routines/solve_routines.jl

```diff
@@ -130,8 +136,140 @@ function run_sim_model!(sim::Simulation; verbose::Bool = false, kwargs...)
             @assert stage.executions == stage.execution_count
             stage.execution_count = 0 # reset stage execution_count
         end
+        
+    end
+    if (:no_dict in keys(kwargs)) == true
```

Comment:
I see that you are controlling the return type of the function with a keyword arg.  I'll suggest that it might be cleaner to accept an optional dict as a parameter defaulting to nothing.  If not nothing, fill it in.  But that's more style.